internal/bminventory/inventory.go

```diff
@@ -629,6 +636,31 @@ func verifyMinimalOpenShiftVersionForSingleNode(requestedOpenshiftVersion string
 	}
 	return nil
 }
+
+func (b *bareMetalInventory) getNewClusterCPUArchitecture(newClusterParams *models.ClusterCreateParams) (string, error) {
+	if newClusterParams.CPUArchitecture == "" {
+		// Empty value implies x86_64
+		return common.DefaultCPUArchitecture, nil
+	}
+
+	cpuArchitectures, err := b.versionsHandler.GetCPUArchitectures(*newClusterParams.OpenshiftVersion)
+	if err != nil {
+		return "", err
+	}
+	for _, cpuArchitecture := range cpuArchitectures {
+		if cpuArchitecture == newClusterParams.CPUArchitecture {
+			return cpuArchitecture, nil
+		}
+	}
+
+	if newClusterParams.CPUArchitecture != common.DefaultCPUArchitecture {
```

Comment:
which case does this `if` covers? 
is there a case that `newClusterParams.CPUArchitecture` is not inside `cpuArchitectures` and we still don't return an error?

Reply:
This validation is for a case in which a non-default architecture is requested, but wasn't found in available architectures list. So, it is not relevant for default architecture as x86_64 is available regardless (for backwards compatibility) - returned in line 661.